### PR TITLE
Extract themed css

### PR DESCRIPTION
**Pull request by umaar**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [ ] Unit or Functional tests are included in the PR

**Description:**

This PR extracts all the CSS which can be themed into their own folders so fixed CSS and theme CSS lives independently. 

I have resolved the conflicts in master already!

Works towards: https://github.com/dojo/cli/issues/168
Related issues: https://github.com/dojo/grunt-dojo2/pull/184 and https://github.com/dojo/cli-create-theme/pull/1  and https://github.com/dojo/interfaces/pull/86 and https://github.com/dojo/cli/pull/180

### CODE REVIEW

**Comment by tomdye** [comment]
Can you standardise on `themeCss` please. Seen both `cssTheme` and `themeCss`.

**Comment by tomdye** [comment]
@smhigley it's likely that the `themes` folder will move entirely out to `@dojo/themes`

File: src/dialog/Dialog.ts [hunk 1]
```diff
@@ -48,6 +49,7 @@ export interface DialogProperties extends ThemedProperties {
 export const ThemedBase = ThemedMixin(WidgetBase);
 
 @theme(css)
```
[the remark this answers is not in the document]
Comment (umaar) [reply src/dialog/Dialog.ts]: Good eye 👁 removed that one, and did what you said about fixed CSS naming conventions in all files, makes total sense

**Comment by umaar** [comment]
@smhigley @tomdye thank you for your reviews - can I kindly request another review, there has been some extra work and they are more than just find/replace operations so I think it's important to get at least one other set of eyes on this whole thing please.

As for the code review:

> Can you standardise on themeCss please. Seen both cssTheme and themeCss.

We are now using `css` and `fixedCss` and that should hopefully all be consistent.

> On a more abstract level, I worry a little that our css is getting overly split up and spread out, so it's more difficult for a css dev coming into an existing project to get a handle on where styles are located. It may be worth it for the improved tooling, though.
Hide outdated

I share the feeling! Working on this, I've been having feelings that this will complicate it for external contributors in a number of ways:

* How should one know a `dojo create theme` command even exists
* How do new widget creators who wish to create consumable widgets know to structure their CSS in such a way
* It can come across weird that not only is a certain CSS structure needed, but on top of that, it is still required to use `theme decorators` and call `theme` methods (you could argue that only one of those should be required at a minimum)

I guess the solution to these is documentation: but then at the same time I feel the more things to remember, the greater the cognitive overhead. I also feel that concepts/techniques like these work best when they are intuitive.

These are just some personal thoughts that came up while reflecting on the work I've been doing, and the tooling we release could turn out to be very useful! From the start, we did have an open mind to different ideas and this is something which was suggested and most people seemed to like it, so let's use it and give feedback!

**Comment by umaar** [comment]
Hey if anyone has time for a review that'd be useful thank you! Maybe @tomdye or @smhigley or someone else?

File: src/themes/dojo/progress.m.css.d.ts [hunk 2]
```diff
@@ -1,4 +1,3 @@
-export const root: string;
```
[the remark this answers is not in the document]
Comment (umaar) [reply src/themes/dojo/progress.m.css.d.ts]: Ohh I ran `tcm` again and it generated/overwrote the `.d.ts` files and it looks like `src/themes/dojo/progress.m.css` legit has no `root` selector! So I guess that is correct in this case?

File: src/theme/common/icons.m.css [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	speak: none;
+	font-style: normal;
+	font-weight: normal;
+	font-variant: normal;
+	text-transform: none;
+
+	/* Better Font Rendering =========== */
+	-webkit-font-smoothing: antialiased;
+	-moz-osx-font-smoothing: grayscale;
+}
```
[the remark this answers is not in the document]
Comment (umaar) [reply src/theme/common/icons.m.css]: Actually my "workflow" was doing each selector one by one. And everything which was "left over" I just left in that file - but I've put them all into `src/theme/common` now!

File: src/slidepane/SlidePane.ts [hunk 4]
```diff
@@ -218,7 +219,7 @@ export default class SlidePane<P extends SlidePaneProperties = SlidePaneProperti
 			align = Align.left,
 			open = false
 		} = this.properties;
-		const alignCss: {[key: string]: any} = css;
+		const alignCss: {[key: string]: any} = fixedCss;
```
[the remark this answers is not in the document]
Comment (umaar) [reply src/slidepane/SlidePane.ts]: I'm very grateful for your explanations and clarifications on what I need to do 🙂

Fixed!

File: src/listbox/ListboxOption.ts [hunk 5]
```diff
@@ -19,7 +21,7 @@ export interface ListboxOptionProperties extends ThemedProperties {
 
 export const ThemedBase = ThemedMixin(WidgetBase);
 
-@theme(css)
+@theme(fixedCss)
```
[the remark this answers is not in the document]
Comment (umaar) [reply src/listbox/ListboxOption.ts]: Nice, fixed, and did a project search to make sure that's not happening elsewhere

File: src/listbox/Listbox.ts [hunk 6]
```diff
@@ -156,10 +157,10 @@ export default class Listbox<P extends ListboxProperties = ListboxProperties> ex
...
 	protected getOptionClasses(active: boolean, disabled: boolean, selected: boolean) {
 		return [
-			css.option,
-			active ? css.activeOption : null,
-			disabled ? css.disabledOption : null,
-			selected ? css.selectedOption : null
+			fixedCss.option,
+			active ? fixedCss.activeOption : null,
+			disabled ? fixedCss.disabledOption : null,
+			selected ? fixedCss.selectedOption : null
```
[the remark this answers is not in the document]
Comment (umaar) [reply src/listbox/Listbox.ts]: Really good observation. When I 'traced' those classes, I saw it was passed to a `classes` property without any `this.theme` going on. What I missed? = Those `classes` are  actually passed to `ListboxOption` which _does_ in turn use `this.theme`. 
Thank you!

Fixed.

File: src/combobox/styles/comboBox.m.css [hunk 7]
```diff
@@ -1,20 +1,19 @@
 @import '../../common/styles/variables.css';
 
-.root {
-	position: relative;
-	display: inline-block;
-	box-sizing: border-box;
-	width: 200px;
-}
+/* Are these styles which the other imported widgets refer to used? */
```
[the remark this answers is not in the document]
Comment (umaar) [reply src/combobox/styles/comboBox.m.css]: Thank you for explaining and for catching those comments! Perfect have removed those selectors

File: src/calendar/tests/functional/Calendar.ts [hunk 8]
```diff
@@ -3,7 +3,9 @@ const { assert } = intern.getPlugin('chai');
 
 import { Remote } from 'intern/lib/executors/Node';
 import keys from '@theintern/leadfoot/keys';
-import * as css from '../../styles/calendar.m.css';
+import * as css from '../../../theme/calendar/calendar.m.css';
+import * as cssCell from '../../../theme/calendar/calendarCell.m.css';
+import * as calendarCellCss from '../../../theme/calendar/calendarCell.m.css';
```
[the remark this answers is not in the document]
Comment (umaar) [reply src/calendar/tests/functional/Calendar.ts]: Good eye 👁 fixed

File: src/slidepane/SlidePane.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (smhigley) [reply src/slidepane/SlidePane.ts]: Thanks! This was totally our fault for messing up the `SlidePane` classes and never noticing, so thanks for fixing it :D

**Comment by umaar** [comment]
@smhigley thank you for a detailed and thorough code review! And all of your explanations were really useful!

I've replied to each comment just so you know nothing has been missed out. Instead of reviewing everything what do you think about reviewing only the changes since you last gave a review? If that's ok then this URL: https://github.com/umaar/widgets/compare/b3a08899bc8762f1a3dbbd724de32ac1d68e38b7...umaar:extract-themed-css should show only the new changes since the last code review. Also there are some test changes which might look unrelated (https://github.com/umaar/widgets/compare/b3a08899bc8762f1a3dbbd724de32ac1d68e38b7...umaar:extract-themed-css#diff-9a18ca9dddd60ae18505da845776ae27L175) but the issue is the tests were incorrect previously!

Thank you again 🌞

**Comment by umaar** [comment]
Oh and also rebased with master!

File: src/themes/dojo/progress.m.css.d.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (smhigley) [reply src/themes/dojo/progress.m.css.d.ts]: You're completely right :). I didn't originally notice that this was in `themes/dojo` and not in `theme/progress`.

**Comment by umaar** [comment]
@smhigley thank you so much! Ok I think this is finally the last set of changes that have been made so maybe if you or @tomdye are able to give a quick sanity check, that'd be good! RE: the changes: I did a rebase with master - and during that process, there were quite a few rounds of patches I applied as there were quite a few conflicts, but it should be all good since the tests pass!

**Pull request merged**